core/solver/bicg.cpp

```diff
@@ -124,8 +125,9 @@ void Bicg<ValueType>::apply_impl(const LinOp* b, LinOp* x) const
 
 
 template <typename ValueType>
-void Bicg<ValueType>::apply_dense_impl(const matrix::Dense<ValueType>* dense_b,
-                                       matrix::Dense<ValueType>* dense_x) const
+template <typename VectorType>
+void Bicg<ValueType>::apply_dense_impl(const VectorType* dense_b,
+                                       VectorType* dense_x) const
```

Comment:
This is not distributed yet, right? (the `Vector` is still dense and it's the default dispatch). Then maybe there's no need to change the template type yet?